### PR TITLE
HIVE 20958: clean code at Hive common module

### PR DESCRIPTION
**Pull request by b-slim**
https://issues.apache.org/jira/browse/HIVE-20958

### CODE REVIEW

File: common/src/java/org/apache/hadoop/hive/common/metrics/metrics2/CodahaleMetrics.java [hunk 1]
```diff
@@ -477,10 +452,8 @@ private boolean initMetricsReporter() {
       default:
         LOGGER.warn("Unhandled reporter " + reporter + " provided.");
       }
-      if (codahaleReporter != null) {
```
Comment (kgyrtkirk) [inline common/src/java/org/apache/hadoop/hive/common/metrics/metrics2/CodahaleMetrics.java]: I think on the default branch; codahaleReporter will remain null

File: common/src/java/org/apache/hadoop/hive/common/type/Decimal128.java [hunk 2]
```diff
@@ -1100,7 +1097,7 @@ public Decimal128 subtractDestructive(Decimal128 right, short scale) {
    */
   public static void multiply(Decimal128 left, Decimal128 right,
       Decimal128 result, short scale) {
-    if (result == left || result == right) {
```
Comment (kgyrtkirk) [inline common/src/java/org/apache/hadoop/hive/common/type/Decimal128.java]: I think this is not righ; the exception was intended to obstruct possible bad preference passing; Object.equals is compare by content.

File: common/src/java/org/apache/hadoop/hive/common/type/Decimal128.java [hunk 3]
```diff
@@ -1210,7 +1207,7 @@ public void multiplyDestructive(Decimal128 right, short newScale) {
    */
   public static void divide(Decimal128 left, Decimal128 right,
       Decimal128 quotient, short scale) {
-    if (quotient == left || quotient == right) {
```
Comment (kgyrtkirk) [inline common/src/java/org/apache/hadoop/hive/common/type/Decimal128.java]: same as above ; keep  '=='?

File: common/src/java/org/apache/hadoop/hive/common/type/SignedInt128.java [hunk 4]
```diff
@@ -683,7 +683,7 @@ public void subtractDestructive(SignedInt128 right) {
    */
   public static void multiply(SignedInt128 left, SignedInt128 right,
       SignedInt128 result) {
-    if (result == left || result == right) {
```
Comment (kgyrtkirk) [inline common/src/java/org/apache/hadoop/hive/common/type/SignedInt128.java]: same here ==

File: common/src/java/org/apache/hadoop/hive/common/type/SignedInt128.java [hunk 5]
```diff
@@ -667,7 +667,7 @@ public void subtractDestructive(SignedInt128 right) {
     }
 
     byte signum = UnsignedInt128.difference(this.mag, right.mag, this.mag);
-    this.negative = (signum > 0 ? this.negative : !this.negative);
```
Comment (kgyrtkirk) [inline common/src/java/org/apache/hadoop/hive/common/type/SignedInt128.java]: great idea! how did you notice this?

File: common/src/java/org/apache/hadoop/hive/common/type/SignedInt128.java [hunk 6]
```diff
@@ -768,7 +768,7 @@ public int divideDestructive(int right) {
    */
   public static void divide(SignedInt128 left, SignedInt128 right,
       SignedInt128 quotient, SignedInt128 remainder) {
-    if (quotient == left || quotient == right) {
```
Comment (kgyrtkirk) [inline common/src/java/org/apache/hadoop/hive/common/type/SignedInt128.java]: ==/equals

File: common/src/java/org/apache/hive/common/util/HiveStringUtils.java [hunk 7]
```diff
@@ -1004,7 +997,6 @@ public static int findTrailingSpaces(byte[] bytes, int start, int length) {
     for (numSpaces = 0; numSpaces < length; ++numSpaces) {
       int curPos = start + (length - (numSpaces + 1));
       if (isAscii(bytes[curPos]) && Character.isWhitespace(bytes[curPos])) {
-        continue;
```
Comment (kgyrtkirk) [inline common/src/java/org/apache/hive/common/util/HiveStringUtils.java]: this is now an empty branch

File: common/src/test/org/apache/hadoop/hive/common/metrics/TestLegacyMetrics.java [hunk 8]
```diff
@@ -78,7 +82,7 @@ public void testMetricsMBean() throws Exception {
     assertTrue(resetFound);
 
     // add metric with a non-null value:
-    Attribute attr = new Attribute("fooMetric", Long.valueOf(-77));
```
Comment (kgyrtkirk) [inline common/src/test/org/apache/hadoop/hive/common/metrics/TestLegacyMetrics.java]: I believe L suffix is better than cast

File: common/src/test/org/apache/hadoop/hive/common/type/HiveDecimalTestBase.java [hunk 9]
```diff
@@ -88,7 +86,6 @@ public BigDecimal randHiveBigDecimalNormalRange(Random r, String digitAlphabet)
     boolean negated = false;
     if (r.nextBoolean()) {
       bigInteger = bigInteger.negate();
-      negated = true;
```
Comment (kgyrtkirk) [inline common/src/test/org/apache/hadoop/hive/common/type/HiveDecimalTestBase.java]: this assignment goes away; but it seems like "negated" is an unreferenced local variable....doesn't it?

File: common/src/test/org/apache/hadoop/hive/common/type/HiveDecimalTestBase.java [hunk 10]
```diff
@@ -158,7 +152,6 @@ public BigInteger randHiveBigInteger(Random r, String digitAlphabet) {
...
```
Comment (kgyrtkirk) [inline common/src/test/org/apache/hadoop/hive/common/type/HiveDecimalTestBase.java]: there are a coubple more unrefderenced negated candidates....

File: common/src/test/org/apache/hadoop/hive/common/type/TestHiveDecimalOrcSerializationUtils.java [hunk 11]
```diff
@@ -116,11 +116,9 @@ private void testSerializationUtilsWriteRead(String string) {
         // System.out.println("TEST_FAST_SERIALIZATION_UTILS_WRITE_BIG_INTEGER bytesExpected " + displayBytes(bytesExpected, 0, bytesExpected.length));
       }
       // Deserialize and check...
-      which = 1;
```
Comment (kgyrtkirk) [inline common/src/test/org/apache/hadoop/hive/common/type/TestHiveDecimalOrcSerializationUtils.java]: unreferenced variable?

File: common/src/java/org/apache/hadoop/hive/common/type/Decimal128.java (hunk 2 above)
Comment (b-slim) [reply common/src/java/org/apache/hadoop/hive/common/type/Decimal128.java]: Good catch. Will add a test to cover this case and make sure that it is enforced.

**Comment by b-slim** [comment]
@kgyrtkirk thanks for taking a look, i will add some check style rules about import order. Current code base has no unique style which is bad IMO some files start with java.* and some other java.* is grouped to the end. I think most of the new projects and IDE prefer grouping JDK stuff at the bottom with a different group for static imports. What you think ?

**Pull request closed without merging**